Module-generation/frontmatter.py

```diff
@@ -14,89 +11,63 @@
         getModuledata = open(moduleFile,"r")
         readModuleData=getModuledata.read()
         parseData=json.loads(readModuleData)
-        
-    
         moduleName=parseData['id']
         moduleDescription=parseData['description']
         
-       
-        
+        moduleLogo="logo.jpg"
+        moduleReadme=DIR+folders+"/README.md"
 
+        os.mkdir(DST+moduleName)
+        IndexMd=open(DST+moduleName+"/index.md","a+")
+        IndexMd.write('---\n')
+        IndexMd.write('posttype: "module" \n')
+        IndexMd.write('title: '+moduleName+'\n')
+        IndexMd.write('description: "'+moduleDescription+'"\n')
+
+        if(os.path.isfile(moduleLogo)):
+            print("yes")
+            IndexMd.write('cover: "./logo.jpg"'+'\n')
+            sourceImage=open(moduleLogo, "rb")
+            readImage=sourceImage.read()
+            ImageFile = open(DST+moduleName+"/logo.jpg", "wb+")    
+            ImageFile.write(readImage)
+            ImageFile.close()
+            sourceImage.close()
+
+        Tags=[]
         if "isGameplay" in parseData:
-            gameplay=parseData['isGameplay']
-        else:
-            gameplay=False
+            if (parseData['isGameplay']):
```

Comment:
not sure if I'm missing something here, but do we expect `parseData['isGameplay']` to fail if directly before we checked that `isGameplay` is in `parseData`? :thinking:

Reply:
we have to check for that because module.txt doesn't have all the keys , if the key is not present how it will parse data from module.txt, so first I have to check for the keys are present or not , if yes then parse data else nothing